### PR TITLE
Draw TMM of my potential move

### PR DESCRIPTION
**Pull request by pakfront**
This addresses RFE https://github.com/MegaMek/megamek/issues/3662
This moves drawing MASC and Supercharger roll targets into a method combined with new functionality of drawing the potential Target Movement Modifier (TMM) on the final step of a planned movement.
Previously the MASC rolls drew every step, which was cluttered and unhelpful, so combining it with final TMM made sense.
Capture
Capture

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/boardview/StepSprite.java [hunk 1]
```diff
@@ -465,38 +465,32 @@ private void drawMovementCost(MoveStep step, boolean isLastStep,
...
-            costStringBuf.append("*");
+            costStringBuf.append('*');
         }
 
         // If the step is past danger, mark that.
         if (step.isPastDanger()) {
-            costStringBuf.insert(0, "(");
-            costStringBuf.append(")");
+            costStringBuf.insert(0, '(');
+            costStringBuf.append(')');
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/boardview/StepSprite.java:484]: Why the swap from double to single quotes?

File: megamek/src/megamek/client/ui/swing/boardview/StepSprite.java [hunk 2]
```diff
@@ -465,38 +465,32 @@ private void drawMovementCost(MoveStep step, boolean isLastStep,
...
 
         EntityMovementType moveType = step.getMovementType(isLastStep);
-        if ((moveType == EntityMovementType.MOVE_VTOL_WALK)
-                || (moveType == EntityMovementType.MOVE_VTOL_RUN)
-                || (moveType == EntityMovementType.MOVE_VTOL_SPRINT)
-                || (moveType == EntityMovementType.MOVE_SUBMARINE_WALK)
-                || (moveType == EntityMovementType.MOVE_SUBMARINE_RUN)) {
-            costStringBuf.append("{").append(step.getElevation())
-                    .append("}");
+        if ((moveType == EntityMovementType.MOVE_VTOL_WALK) || (moveType == EntityMovementType.MOVE_VTOL_RUN) || (moveType == EntityMovementType.MOVE_VTOL_SPRINT) || (moveType == EntityMovementType.MOVE_SUBMARINE_WALK) || (moveType == EntityMovementType.MOVE_SUBMARINE_RUN)) {
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/boardview/StepSprite.java:488]: Really not convinced this is easier to read.

File: megamek/src/megamek/client/ui/swing/boardview/StepSprite.java [hunk 3]
```diff
@@ -510,38 +504,51 @@ private void drawMovementCost(MoveStep step, boolean isLastStep,
         graph.drawString(costString, costX, stepPos.y + 39);
         graph.setColor(col);
         graph.drawString(costString, costX - 1, stepPos.y + 38);
+    }
+
+    private void drawTMMAndRolls(MoveStep step, boolean isLastStep, boolean jumped, Game game,
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/boardview/StepSprite.java:509]: Apparently, the parameter "isLastStep" is unused. Let's figure out what's up with that.

**Review by NickAragua — CHANGES_REQUESTED** [review]
Need a little cleanup, but looks pretty good overall.

File: megamek/src/megamek/client/ui/swing/boardview/StepSprite.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/boardview/StepSprite.java:484]: Slightly better performance, which adds up the more swaps are done.

File: megamek/src/megamek/client/ui/swing/boardview/StepSprite.java (hunk 2 above)
Comment (pakfront) [reply megamek/src/megamek/client/ui/swing/boardview/StepSprite.java:488]: IDE did that to me. I will revert.

**Review by NickAragua — APPROVED** [review]

**Pull request merged**